app/scripts/modules/kubernetes/src/pipeline/stages/runJob/runJobStage.js

```diff
@@ -76,47 +101,96 @@ module.exports = angular.module('spinnaker.kubernetes.pipeline.stage.runJobStage
         this.accounts = accounts;
       });
 
-    this.stage.cloudProvider = 'kubernetes';
-    this.stage.application = $scope.application.name;
 
-    if (!this.stage.credentials && $scope.application.defaultCredentials.kubernetes) {
-      this.stage.credentials = $scope.application.defaultCredentials.kubernetes;
+    function buildImageId(image) {
```

Comment:
This is just a label for the UI, right? I wouldn't want to maintain this logic in two places.

It feels like you could have an Angular filter that takes an image and returns the string, and then use that filter in the server group configuration dialogue and here.

Reply:
Can you point me toward an example of something like this?

Reply:
Yes - I was thinking you could define a custom filter named `kubernetesImageId` (or something) and include it inside your `ui-select`:

`<span ng-bind-html="container.imageDescription | kubernetesImageId | highlight: $select.search"></span>`

You can define a filter like this: https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/presentation/robotToHumanFilter/robotToHuman.filter.ts